\documentclass[12pt]{article}
\usepackage{fullpage,amsmath, mathtools}
\usepackage{setspace}
\usepackage{graphicx, psfrag, amsfonts}
\usepackage{natbib}
\usepackage{amsthm}
\usepackage{multirow}
\usepackage{hhline}


\newtheorem{theorem}{Theorem}
\newtheorem{lem}{Lemma}
\newtheorem{corollary}{Corollary}
\newtheorem{definition}{Definition}
\newtheorem{assumption}{Assumption}
\newtheorem{proposition}{Proposition}


% ***************************************************************
% JUHEE: just comment out this line to get the figures back :-)
%\renewcommand{\includegraphics}[2][1]{FIGURE HERE}


\def\bth{\mbox{\boldmath $\theta$}}
\def\bmu{\mbox{\boldmath $\mu$}}
\def\bgam{\mbox{\boldmath $\gamma$}}
\def\bsig{\mbox{\boldmath $\sigma$}}
\def\bomega{\mbox{\boldmath $\omega$}}
\def\bdeta{\mbox{\boldmath $\eta$}}
\def\bpi{\mbox{\boldmath $\pi$}}
\def\bbeta{\mbox{\boldmath $\beta$}}
\newcommand{\pit}{\widetilde{\pi}}
\def\bpit{\mbox{\boldmath $\pit$}}

\newcommand{\thethe}{ } % {{\bf THE}}  % YUAN & JUHEE:
   %% in many places i felt we should drop the definte article "the".
   %% To see all of them just replace this command.
\newcommand{\bY}{\mbox{\boldmath $Y$}}
\newcommand{\by}{\mbox{\boldmath $y$}}
\newcommand{\bx}{\mbox{\boldmath $x$}}
\newcommand{\bn}{\mbox{\boldmath $n$}}
\newcommand{\bN}{\mbox{\boldmath $N$}}
\newcommand{\bc}{\mbox{\boldmath $c$}}
\newcommand{\bd}{\mbox{\boldmath $d$}}
\newcommand{\tbc}{\widetilde{\mbox{\boldmath $c$}}}
\newcommand{\bchat}{\widehat{\mbox{\boldmath $c$}}}
\newcommand{\bZ}{\mbox{\boldmath $Z$}}
\newcommand{\bL}{\mbox{\boldmath $L$}}
\newcommand{\bM}{\mbox{\boldmath $M$}}
\newcommand{\bz}{\mbox{\boldmath $z$}}
\newcommand{\bell}{\mbox{\boldmath $\ell$}}
\renewcommand {\bpi} {\mbox{\boldmath $\pi$}}
\newcommand{\No} {\text{N}}
\newcommand {\bg} {\mbox{\boldmath $g$}}
\newcommand{\Bern} {\text{Bern}}

\newcommand{\bV}{\mbox{\boldmath $V$}}
\newcommand{\bv}{\mbox{\boldmath $v$}}
\newcommand{\brr}{\mbox{\boldmath $r$}}
\newcommand {\bh} {\mbox{\boldmath $h$}}

\newcommand{\wt}{\mbox{$\tilde{w}$}}
\newcommand{\Mb}{\mbox{$\bar{M}$}}
\newcommand{\bwt}{\mbox{\boldmath $\tilde{w}$}}
\newcommand{\bw}{\mbox{\boldmath $w$}}
\newcommand{\bp}{\mbox{\boldmath $p$}}
\newcommand{\pt}{\mbox{$\tilde{p}$}}

\newcommand{\btau}{\bm{\tau}}
\newcommand{\bnu}{\bm{\nu}}
\newcommand{\Sbar}{\overline{S}}
\newcommand{\Var} {\text{Var}}

\newcommand{\Ga}{\mbox{Gamma}}
\newcommand{\Dir}{\mbox{Dir}}
\newcommand{\Ber}{\mbox{Ber}}
\newcommand{\Be}{\mbox{Be}}
\newcommand{\Unif}{\mbox{Unif}}
\newcommand{\Nor}{\mbox{N}}
\newcommand{\Binom}{\mbox{Bin}}
\newcommand{\Multi}{\mbox{Multinomial}}
\newcommand{\Exp}{\mbox{E}}


\newcommand{\Gprime}{G^\prime}
\newcommand{\Normal}{\mathrm{N}}
\newcommand{\Poi}{\mathrm{Poi}}
\renewcommand{\th}{\theta}
\newcommand{\thstar}{\theta^\star}
\newcommand{\sig}{\sigma}
\newcommand{\sigsq}{\sig^2}

\newcommand{\iid}{\stackrel{iid}{\sim}}
\newcommand{\indep}{\stackrel{indep}{\sim}}

\usepackage[dvipsnames,usenames]{color}
\newcommand{\bch}{\color{blue}\it}
\newcommand{\bchm}{\color{red}}
\newcommand{\ech}{\rm\color{Black}\normalsize}
\newcommand{\note}[1]{\color{Orange} Note \color{Black} \footnote{\color{Orange}\rm #1 \color{Black}}}

\newcommand{\yy}{\color{magenta}}
\newcommand{\jj}{\color{Black}\normalsize}
\newcommand{\yjnote}[1]{\color{blue} Note \color{Black} \footnote{\color{Brown}\rm #1 \color{Black}}}

\newcommand{\cut}{\color{green}[cut] \color{Black}  }
\renewcommand{\split}{\color{green}[split] \color{Black}  }

\newcommand{\hh}{\color{Mahogany}}




\begin{document}

\centerline {\bf Tumor Purity}
\vskip .5in

\today

\begin{enumerate}
\item Goal:  develop a statistical method to infer tumor purity using DNA-seq data.

\vspace*{0.1in}


\item Notation
\begin{enumerate}
\item Loci associated with single nucleotide variants (SNVs): $s=1, \ldots, S$
\item $M_{s}$: tumor cell copy number at locus $s$ from a copy number caller  --- \textbf{Observed}
\item $N_{ks}$: total number of reads at locus $s$ in sample $k$ where $k=0$ for normal and 1 for tumor  --- \textbf{Observed}
\item $n_{ks}$:  number of reads indicating variants at locus $s$ in sample $k$  --- \textbf{Observed}
\end{enumerate}


\item Likelihood
\begin{itemize}
\item $n_{1s}$:  Define sampling models for $n_{1s}$ conditional on $N_{1s}$ and $M_s$.
\begin{eqnarray*}
n_{1s} \mid N_{1s}, p_{s} &\sim& \mbox{Binom}(N_{1s}, p_{s}),  
\end{eqnarray*}
where 
\begin{description}
\item[$p_{s}$:] the expected proportion of reads indicating mutation at locus $s$
\end{description}

We formulate $p_s$ as follows:
\begin{eqnarray*}
p_s =\frac{\mu \times v_s  \times M_s}{(1-\mu)\times 2 + \mu \times M_s}.
\end{eqnarray*}
Reasoning:  
\begin{itemize}
\item $\mu$: tumor cell fraction (tumor purity)
\item no copy number variant in normal cells  $\Rightarrow$ the expected total number of alleles at locus $s$ becomes the average of normal cell copy number (2) and  tumor cell copy number ($M_s$) by weights $(1-\mu)$ (normal cell fraction) and $\mu$.
\item $v_s$: the proportion of mutated alleles at locus $s$
\end{itemize}

\item $\log(N_{1s}/N_{0s})$: logR
$$
\log(N_{1s}/N_{0s}) \mid M_s, \sigma^2, \mu \sim \Nor\left(\log\left(\frac{(1-\mu)2 + \mu M_s}{2}\right) + \phi, \sigma^2\right).
$$
Note that if $M_s=2$, the mean of $\log(N_{1s}/N_{0s})$ becomes $\phi$, some adjustment by difference in sequencing depth. 
\end{itemize}





\item Priors
\begin{enumerate}
\item $v_s$
\begin{align*}
  v_s | G &\sim G \\
  G &\sim \text{DP}(\alpha,G_0),
\end{align*}
where $G_0 = \Be(a_v, b_v)$ and $\alpha$ is the total mass parameter.

\item $\mu$
$$
\mu \sim \Be(a_\mu, b_\mu).
$$

\item $\phi$
  $$
  \phi \sim N(m_\phi, s^2_\phi)
  $$

\item $\sigma^2$
  $$
  \sigma^2 \sim IG(a_\sigma,b_\sigma)
  $$

\end{enumerate}
\end{enumerate}



\end{document}



